ODIN_II/SRC/partial_map.cpp

```diff
@@ -186,13 +186,22 @@ void partial_map_node(nnode_t *node, short traverse_number, netlist_t *netlist)
 			break;
 
 		case ADD:
-			if (hard_adders && node->bit_width >= min_threshold_adder){
-				// Check if the size of this adder is greater than the hard vs soft logic threshold
-					instantiate_hard_adder(node, traverse_number, netlist);
-			}else{
-				instantiate_add_w_carry(node, traverse_number, netlist);
+			if (true == mixer->mixHardBlocksOfType(HardBlocksOptimizationTypesEnum::MULTIPLIERS) ||
+				true == mixer->mixHardBlocksOfType(HardBlocksOptimizationTypesEnum::ADDERS)  )
+			{
+				mixer->takeNoteOfAPotentialHardBlockNode(node,HardBlocksOptimizationTypesEnum::ADDERS);
+				break;
+			} else
+			{
+				if (hard_adders && node->bit_width >= min_threshold_adder){
+					// Check if the size of this adder is greater than the hard vs soft logic threshold
+						instantiate_hard_adder(node, traverse_number, netlist);
+				}else{
+					instantiate_add_w_carry(node, traverse_number, netlist);
+				}
+				break;
 			}
-			break;
```

Comment:
The first if statement is gonna fall through without this break, avoid fallthrough with the break at the end of your case or put the non breaking logic here at the end and explicitly fallthrough by putting at least a //fallthrough comment

Reply:
I "think" there's no fall through here because the true part of the statement is terminated with break, so is the else part. Alternatively, for readability purposes I can introduce a {} around if, but it seemed unreasonable to introduce {} for a single if statement.

Reply:
We always use the bracket wether it be for one line or Multiple. Just the way it was written to begin with, some files seem to stray away from that but that's because no on was enforcing the standard.

Reply:
To stick with the rest of the file it should look something like this
```c

		case ADD:
			if (mixer->can_be_hardened(MULTIPLY) || mixer->can_be_hardened(ADD) )
			{
				mixer->flag_for_hardenning(node,ADD);
			}
                        else
			{
				if (hard_adders && node->bit_width >= min_threshold_adder)
                                {
					instantiate_hard_adder(node, traverse_number, netlist);
				}
                                else
                                {
					instantiate_add_w_carry(node, traverse_number, netlist);
				}
			}
                        break;
```

Reply:
Thank you, fixed! I will ship the changes in the next commit